authentication/views.py

```diff
@@ -56,6 +56,20 @@ def post(self, request):
                 'email': email
             })
 
+class RelateUserToCompany(APIView):
+    """Binding user to company and inserting linked company's id into token."""
+    permission_classes = (IsAuthenticated,)
+    
+    def post(self, request):
+        user_id = request.user.user_id
+        company_id = request.data['company_id']
```

Comment:
Are you sure, that `company_id` will always be present in the request data?Consider using `request.data.get('company_id'`) to avoid `KeyError` if `company_id` is missing. You can then check if `company_id` is `None` and return a `400 Bad Request` response if necessary.

Reply:
We catch KeyError and raise NotAuthenticated exception, which allow adress possible absence of company_id or user_id in token payload